### ISSUE TITLE
Translation strings used in .pot and .po files, from domains 'qts', 'qtranslate' and no domain.

### ISSUE THREAD

**Issue by pedro-mendonca** [not-only-code/qtranslate-slug#95]
I've been using Poedit to update .pot catalog and portuguese pt_PT.po files.

When I include __() and _e() keywords in Poedit properties, it parses all php and include both 'qts', 'qtranslate' and some __('string') without 'text-domain'.
I believe that it's only needed for the actual translation the strings from 'qts' domain, as 'qtranslate' reads the main qtranslate (or fork) plugin and the the strings without domain tries to read the WordPress base language files.

What should I as a translator do?
a) Translate all strings, from 'qts', 'qtranslate' and without domain, even knowing that it won't be used justo to have 100% strings translation report.
b) Manually delete those strings that don't belong to 'qts' domain.
c) Change all strings to 'qts', overriding any other available translations in other plugins and WordPress base language files.
d) Use some kind of filtering to exclude any kind of strings from other domains than 'qts' when generating .pot catalog and .po files.

**Comment by LC43** [comment]
Hi @pedro-mendonca , very good question. I myself been wondering the same. To be honest there's no good tool to generate the .pot file using only a textdomain. Thinking on your perfectly laid out possibilities, maybe we should opt to use d). We can do that by  using the  "context",  as in `_x('Language List',"don't translate this","qts");`  or we can use a [notice ](https://developer.wordpress.org/themes/functionality/internationalization/#descriptions)

**Comment by pedro-mendonca** [comment]
Hi @LC43 , I think a notice won't help to filter out automatically the not-to-translate strings as you cannot use it to exclude anything in Poedit.
Your suggestion of use contexted keywords as in _x('Language List',"don't translate this","qts"); might just work.
You might keep strings to translate in __() and _e() and the _x() function can be excluded from keyword list in .pot and .po files. I believe this will work right.
You can still add a notice like "don't translate this", "translated-by-main-qtranslate", "WordPress base string", just clarify that it's really not needed to translate, but it won't be included in the automatically parsed list because of _x() function.
I'll learn how to use this _x() and try it to see if it's a way to go.

**Comment by pedro-mendonca** [comment]
@LC43 as far as I could check, the _x() / _ex() functions could work with a context note, but the not must exist also in 'qtranslate-XX.po/.mo. It doesn't make sense because it's out of control of qtranslate-slug work.

Examples:
__( 'Settings' /\* already translated in main WordPress _/ )
_x( 'Settings', 'CONTEXT', 'qtranslate' /_ already translated in main qTranslate plugin */ )

_x() can really be filtered out, by simply not includind it in the poedit keywords, but you will need the main plugin qtranslate-xx_XX.po language file to include msgctxt as long as msgid and msgstr. In theory it's out of range of qtranslate-slug developers.
Makes more sense to me in the future that the qtranslate-slug would be included as an option within the main qTranslate plugin, as you actually substitute some code from the widget, etc.

For now, the only solution I can think of is really use the plugin text-domain 'qts' to all strings.
Of course it's redundant, but it's only made once, and you just have to take care to use the same as the originals in 'qtranslate'.

Another argument in favour of future merging with qTranslate-x (suggestion) is that you're granted that those codebits like 'Text only'; 'Image only'; 'Text and Image'; 'Dropdown Box' are all coded just once in one plugin, without the need to duplicate it in a dependent one.
This is, of course, a mere suggestion because of this problem we found in external tex-domains, but I think no one better than you could think of how to make it an option included in the main one :)

What do you thing about all this?
Shall we go for 'qts' in every string?
Would you think of slug translation as an option for the recent qTranslate-X future merge and emerging platform?

Best regards
Pedro

**Comment by LC43** [comment]
On Thursday, January 15, 2015, Pedro Mendonça notifications@github.com
wrote:

> @LC43 https://github.com/LC43 as far as I could check, the _x() / _ex()
> functions could work with a context note, but the not must exist also in
> 'qtranslate-XX.po/.mo. It doesn't make sense because it's out of control of
> qtranslate-slug work.
> 
> Example:
> _x( 'Settings', 'CONTEXT', 'qtranslate' /\* no domain, use WordPress base
> translation */ )
> 
> _x() can really be filtered out, by simply not includind it in the poedit
> keywords, but you will need the main plugin qtranslate-xx_XX.po language
> file to include msgctxt as long as msgid and msgstr. In theory it's out of
> range of qtranslate-slug developers.
> Makes more sense to me in the future that the qtranslate-slug would be
> included as an option within the main qtranslate plugin, as you actually
> substitute some code from the widget, etc.
> 
> For now, the only solution I can think of is really use the plugin
> text-domain 'qts' to all strings.
> 
> Of course it's redundant, but it's only made once, and you just have to
> take care to use the same as the originals in 'qtranslate'.

You're right, good job hunting all this information! Let'd move into this
then, whenever you have time. Im about to work with a client in morocco, so
ill be away for 3 weeks. I wish i could add you to the admins,but carlos is
the only one that can do it. So, meanwhile, i accept any pull request and
push it to main wp.org.

> Another argument in favour of future merging with qTranslate-x
> (suggestion) is that you're granted that those codebits like 'Text only';
> 'Image only'; 'Text and Image'; 'Dropdown Box' are all coded in one plugin,
> without the need to duplicate it in a dependent one.
> This is, of course, a mere suggestion despite the problem we found in
> external tex-domains, but I think no one better than you could think of how
> to make it an option included in the real thing :)

Oh, it has also crossed my mind. The plugin was needed when there wasnt any
active development in qtranslate. There was someone already willing to
merge our plugin with qtranslate-x ( which merged from  qtz, mqt etc) . And
to be honest, we already remove qt filters and use our own: home_url,
qts_use, qts_split, all the url related actions are parse by us too..., the
widget, the menus... So much..

I was even planning to re-write qtrans_convertUrl because there are some
people that need to figure out the url of any post...

> What do you thing about all this?
> Shall we go for 'qts' in every string?
> Would you think of slug translation as an option for the recent
> qTranslate-X future merge and emerging platform?
> 
> Best regards
> 
> Pedro

Im all for it!

**Comment by pedro-mendonca** [comment]
Hi
I'll move every string to 'qts' text domain then. I'll join the language files updated also.
As soon as I have it all, I'll pull it.

About the suggestion of merging, maybe you can keep that in mind just to consider all pros and cons.

For now, let's keep cleaning and improving qts :)

Thank you for your attention on this.
Pedro

**Comment by pedro-mendonca** [comment]
Hi @LC43 

I've just completed all this, transformed every text string to 'qts' text domain.
I found a text "More information about [qTranslate-X;mqTranslate;etc]" that I made the common part translatable.
I've also updated the .pot and pt_PT files
Pull request here: https://github.com/not-only-code/qtranslate-slug/pull/100
For now I think this is Issue is closed.
Maybe the thought about a future merge with qtranslate-x can be kept somewhere else. Any suggestion?
